v-next/hardhat-ignition/src/helpers/hardhat-artifact-resolver.ts

```diff
@@ -0,0 +1,39 @@
+import type { HardhatRuntimeEnvironment } from "@ignored/hardhat-vnext/types/hre";
+import type {
+  Artifact,
+  ArtifactResolver,
+  BuildInfo,
+} from "@ignored/hardhat-vnext-ignition-core";
+
+import { readJsonFile } from "@ignored/hardhat-vnext-utils/fs";
+
+export class HardhatArtifactResolver implements ArtifactResolver {
+  readonly #hre: HardhatRuntimeEnvironment;
+
+  constructor(_hre: HardhatRuntimeEnvironment) {
+    this.#hre = _hre;
+  }
+
+  public async getBuildInfo(
```

Comment:
This has been rework to use the new build info methods.
Previously FQNs and contract names were treated separately.